tests/test_calling_convention_analysis.py

```diff
@@ -20,263 +19,240 @@
-    binary_path = os.path.join(test_location, '..', 'binaries-private', 'cgc_qualifier_event', 'cgc', binary_name)
-    project = angr.Project(binary_path, auto_load_libs=False)
-
-    categorization = project.analyses.FunctionCategorizationAnalysis()
-
-    # tag_manager = categorization.function_tag_manager
-    # print "INPUT:", map(hex, tag_manager.input_functions())
-    # print "OUTPUT:", map(hex, tag_manager.output_functions())
-
+# pylint: disable=missing-class-docstring
+# pylint: disable=no-self-use
+class TestCallingConventionAnalysis(unittest.TestCase):
+    def run_fauxware(self, arch, function_and_cc_list):
+        binary_path = os.path.join(test_location, "tests", arch, "fauxware")
+        fauxware = angr.Project(binary_path, auto_load_libs=False)
 
-def test_fauxware():
+        cfg = fauxware.analyses.CFG()
 
-    amd64 = archinfo.arch_from_id("amd64")
+        for func_name, expected_cc in function_and_cc_list:
+            authenticate = cfg.functions[func_name]
+            _ = fauxware.analyses.VariableRecoveryFast(authenticate)
 
-    args = {
-        'i386': [
-            ('authenticate', SimCCCdecl( archinfo.arch_from_id('i386'), ) ),
-        ],
-        'x86_64': [
-            ('authenticate', SimCCSystemVAMD64( amd64, )
-             ),
-        ],
-    }
+            cc_analysis = fauxware.analyses.CallingConvention(
+                authenticate, cfg=cfg, analyze_callsites=True
+            )
+            cc = cc_analysis.cc
 
-    for arch, lst in args.items():
-        yield run_fauxware, arch, lst
+            assert cc == expected_cc
 
+    def run_cgc(self, binary_name):
+        pass
+        # binary_path = os.path.join(test_location, '..', 'binaries-private', 'cgc_qualifier_event', 'cgc', binary_name)
+        # project = angr.Project(binary_path, auto_load_libs=False)
+        #
+        # categorization = project.analyses.FunctionCategorizationAnalysis()
 
-# def test_cgc():
-def disabled_cgc():
-    # Skip this test since we do not have the binaries-private repo cloned on Travis CI.
+        # tag_manager = categorization.function_tag_manager
+        # print "INPUT:", map(hex, tag_manager.input_functions())
+        # print "OUTPUT:", map(hex, tag_manager.output_functions())
 
-    binaries = [
-        "002ba801_01",
-        "01cf6c01_01",
-    ]
+    def test_fauxware(self):
 
-    for binary in binaries:
-        yield run_cgc, binary
+        amd64 = archinfo.arch_from_id("amd64")
 
+        args = {
+            'i386': [
+                ('authenticate', SimCCCdecl(archinfo.arch_from_id('i386'), )),
+            ],
+            'x86_64': [
+                ('authenticate', SimCCSystemVAMD64(amd64, )
+                 ),
+            ],
+        }
 
-#
-# Full-binary calling convention analysis
-#
+        for arch, lst in args.items():
+            yield self.run_fauxware, arch, lst
 
+    @unittest.skip("Skip this test since we do not have the binaries-private repo cloned on Travis CI.")
```

Comment:
I think this is another one we can add to common.py. `@requires_binaries_private`. It can check if the `binaries-private` repo exists in the same folder as angr, and return skip if not.